src/component/requests.js

```diff
@@ -15,32 +17,69 @@ export default class Requests extends React.Component {
     });
   }
 
+  componentDidMount() {
+    const requests = React.findDOMNode(this);
+    requests.addEventListener('scroll', () => {
+      const fromIndex = Math.ceil(requests.scrollTop / requestElementHeight) - 15;
```

Comment:
I think one performance issue might be calling react render in scroll. That's the only thing that seems to slow it down. I could debounce the scroll event. Do you have any suggestions?